### DISCUSSION TITLE
ignore filename when pipelining rg

### DISCUSSION THREAD

**Discussion by Kyle0923** [BurntSushi/ripgrep#2756] [Q&A]
consider bar.txt with the following content
```
foo
```

`rg foo | rg bar` will match this record because the first search outputs `bar.txt:foo`
in this case, I want to strictly limit my search inside the content, and exclude the filename

Is there a way currently in rg to skip the filename part?

**Comment by BurntSushi** [comment]
The only way is to either suppress the filename from the output with `--no-filename`, or adjust your regex to account for it. For example, `rg '^[^:]+:bar'` won't match `bar.txt:foo`. But of course, this isn't always correct, for example, in cases where the path can contain a colon. It also might need some tweaking to account for line numbers.

**Reply by Kyle0923** [reply]
Thanks for the reply. And thanks for rg, it has been my day-to-day searching tool for a long time

`--no-filename`  is not gonna be very useful because often times, rg is used to search multiple files, losing filename will just lose track of the searching.
for now, i am gonna try with some regexes - `^[^:]+:.*bar` for filename or `^[^:]+:[0-9]+:.*bar` for filename:line_num, but it still has some limitations / costs. For one, I need a longer regex even when my ask is quite simple - grep a word; secondly, the coloring will be a mess, because the regex matches from the filename to `bar`

If I may, I want to propose a `--field n` option which effectively does a `sed -E -e 's/^([^:]+:){n-1}//'` (field is n, so we ignore first n-1 fields) during search while preserving the presentation. So one can do
```
# bar.txt
foo
# baz.txt
foo bar

> rg foo | rg --field 2 bar
baz.txt: foo bar
> rg --line-number foo | rg --field 3 bar
baz.txt:1: foo bar
```

Admittedly, if the filename contains a colon, this will not work, but I think it is rare to have colon in filename anyway

We can even take this one step further.
For example, if I want to grep the PPID (3rd) column in `ps -ef`
I can say `ps -ef | rg --field-separator ' ' --field  3 <pattern>` to grep for the third column (borrowing from awk)
but then in this case, we need to provide a syntax to express from field 3 onward, maybe `--field 3+`
so in our bar.txt, baz.txt example, the cmd becomes `rg foo | rg --field 2+ bar` in case where the file content contains `:`
